enhancements/insights/on-demand-data-gathering.md

```diff
@@ -0,0 +1,255 @@
+
+### Non-Goals
+
+TBD (or maybe N/A)
+
+## Proposal
+
+The enhancement proposal introduces a new Insights `CustomResourceDefinition` type in the OpenShift API that will serve as the definition and trigger to new data gathering on demand. 
+This new CRD will build on and extend the existing `GatherConfig` (see the [Insights config API enhancement](../insights/insights-config-api.md)). The idea is that with each new CR instance of this resource, new data gathering will happen. 
+
+The proposal for the new `datagather.insights.openshift.io` resource is:
+
+```go
+type DataGather struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	Spec DataGatherSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status DataGatherStatus `json:"status"`
+}
+
+type DataGatherSpec struct {
+	// dataPolicy allows user to enable additional global obfuscation of the IP addresses and base domain
+	// in the Insights archive data. Valid values are "None" and "ObfuscateNetworking".
+	// When set to None the data is not obfuscated.
+	// When set to ObfuscateNetworking the IP addresses and the cluster domain name are obfuscated.
+	// When omitted, this means no opinion and the platform is left to choose a reasonable default, which is subject to change over time.
+	// The current default is None.
+	// +optional
+	DataPolicy DataPolicy `json:"dataPolicy,omitempty"`
+	// disabledGatherers is a list of gatherers to be excluded from the gathering. All the gatherers can be disabled by providing "all" value.
+	// If all the gatherers are disabled, the Insights operator does not gather any data.
+	// The particular gatherers IDs can be found at https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md.
+	// Run the following command to get the names of last active gatherers:
+	// "oc get insightsoperators.operator.openshift.io cluster -o json | jq '.status.gatherStatus.gatherers[].name'"
+	// An example of disabling gatherers looks like this: `disabledGatherers: ["clusterconfig/machine_configs", "workloads/workload_info"]`
+	// +optional
+	DisabledGatherers []string `json:"disabledGatherers"`
+}
+
+const (
+	// No data obfuscation
+	NoPolicy DataPolicy = "None"
+	// IP addresses and cluster domain name are obfuscated
+	ObfuscateNetworking DataPolicy = "ObfuscateNetworking"
+	// Data gathering is running
+	Running State = "Running"
+	// Data gathering is completed
+	Completed State = "Completed"
+	// Data gathering failed
+	Failed State = "Failed"
+)
+
+// dataPolicy declares valid data policy types
+// +kubebuilder:validation:Enum="";None;ObfuscateNetworking
+type DataPolicy string
+
+// state declares valid gathering state types
+// +kubebuilder:validation:Enum=Running;Completed;Failed
+type State string
+
+type DataGatherStatus struct {
```

Comment:
Perhaps this should have an object reference which points to the pod that was used to gather the data, so that while it's running, we could if we wanted, check the pod logs, or check the pod for errors

Reply:
> Perhaps this should have an object reference which points to the pod that was used to gather the data, so that while it's running, we could if we wanted, check the pod logs, or check the pod for errors

I like relatedResources for this.  That field is automatically scraped by `oc adm inspect`.  it could also be specific reference, but related resources has some benefit.

Reply:
I added `relatedObjects` attribute. Please check if it's OK.